presto-hive/src/main/java/io/prestosql/plugin/hive/HiveTableHandle.java

```diff
@@ -107,6 +108,35 @@ public HiveTableHandle(
             Optional<HiveBucketFilter> bucketFilter,
             Optional<List<List<String>>> analyzePartitionValues,
             Optional<Set<String>> analyzeColumnNames)
+    {
+        this(
+                schemaName,
+                tableName,
+                tableParameters,
+                partitionColumns,
+                partitions,
+                compactEffectivePredicate,
+                enforcedConstraint,
+                bucketHandle,
+                bucketFilter,
+                analyzePartitionValues,
+                analyzeColumnNames,
+                Optional.empty());
+    }
+
+    public HiveTableHandle(
```

Comment:
Instead of a new constructor, just modify the main one and adjust the callsites that invoke it. There are only a handful. 

BTW, because of how it's done here, there's a theoretical bug (or rather, inconsistency) due to `HiveMetadata.makeCompatiblePartitioning()` wanting to create a copy of the handle by invoking the constructor that doesn't take `constraintColumns`. When that gets called, any value that was set for that field will get lost. It doesn't matter in practice because that happens after the call to `validateScan`, but it's just good hygiene to make it be as consistent as possible.